openedx/core/djangoapps/user_api/models.py

```diff
@@ -25,28 +56,79 @@ class Meta:  # pylint: disable=missing-docstring
         unique_together = ("user", "key")
 
     @classmethod
-    def set_preference(cls, user, preference_key, preference_value):
+    def set_preference(cls, user, preference_key, preference_value, save=True):
+        """Sets the user preference for a given key, creating it if it doesn't exist.
+
+        Arguments:
+            user (User): The user whose preference should be set.
+            preference_key (string): The key for the user preference.
+            preference_value (string): The value to be stored.
```

Comment:
Folks are definitely not always providing a string-- non-string values seem to be converted to a string. Assuming that behavior still works, let's document it. If it doesn't work, let's talk.